src/Files.App/Helpers/Layout/LayoutPreferencesManager.cs

```diff
@@ -664,6 +668,46 @@ public static void SetLayoutPreferencesForPath(string path, LayoutPreferencesIte
 			}
 		}
 
+		private void UpdateLayoutIfRequired(bool isChangingGridSize = false, bool force = false)
+		{
+			var isUsingGridViewWithDifferentSize =
+				LayoutMode is FolderLayoutModes.GridView &&
+				GridViewSize != UserSettingsService.LayoutSettingsService.DefaulIconSizeGridView;
+
+			if (!force && 
+				LayoutMode == UserSettingsService.FoldersSettingsService.DefaultLayoutMode &&
+				!isUsingGridViewWithDifferentSize)
+			{
+				return;
+			}
+
+			switch (UserSettingsService.FoldersSettingsService.DefaultLayoutMode)
+			{
+				case FolderLayoutModes.DetailsView:
+					ToggleLayoutModeDetailsView(true);
+					break;
+				case FolderLayoutModes.ListView:
+					ToggleLayoutModeList(true);
+					break;
+				case FolderLayoutModes.TilesView:
+					ToggleLayoutModeTiles(true);
+					break;
+				case FolderLayoutModes.GridView:
+					IsAdaptiveLayoutEnabled = false;
+					ToggleLayoutModeGridView(isChangingGridSize
+						? GridViewSize
+						: UserSettingsService.LayoutSettingsService.DefaulIconSizeGridView
+					);
+					break;
+				case FolderLayoutModes.ColumnView:
+					ToggleLayoutModeColumnView(true);
+					break;
+				case FolderLayoutModes.Adaptive:
+					ToggleLayoutModeAdaptive();
+					break;
+			}
+		}
```

Comment:
```suggestion
private void UpdateLayoutIfRequired(bool isChangingGridSize = false, bool force = false)
{
	if (!force &&
		LayoutMode == UserSettingsService.FoldersSettingsService.DefaultLayoutMode)
	{
		return;
	}

	switch (UserSettingsService.FoldersSettingsService.DefaultLayoutMode)
	{
		case FolderLayoutModes.DetailsView:
			if (LayoutMode is FolderLayoutModes.DetailsView && GridViewSize != UserSettingsService.LayoutSettingsService.DefaultIconSizeDetailsView)
				ToggleLayoutModeDetailsView(true);
			break;
		case FolderLayoutModes.ListView:
			if (LayoutMode is FolderLayoutModes.ListView && GridViewSize != UserSettingsService.LayoutSettingsService.DefaultIconSizeListView)
				ToggleLayoutModeList(true);
			break;
		case FolderLayoutModes.TilesView:
			if (LayoutMode is FolderLayoutModes.TilesView && GridViewSize != UserSettingsService.LayoutSettingsService.DefaulIconSizeTilesView)
				ToggleLayoutModeTiles(true);
			break;
		case FolderLayoutModes.GridView:
			IsAdaptiveLayoutEnabled = false;

			if (LayoutMode is FolderLayoutModes.GridView && GridViewSize != UserSettingsService.LayoutSettingsService.DefaulIconSizeGridView)
				ToggleLayoutModeGridView(UserSettingsService.LayoutSettingsService.DefaulIconSizeGridView);
			break;
		case FolderLayoutModes.ColumnView:
			if (LayoutMode is FolderLayoutModes.ColumnView && GridViewSize != UserSettingsService.LayoutSettingsService.DefaultIconSizeColumnsView)
				ToggleLayoutModeColumnView(true);
			break;
		case FolderLayoutModes.Adaptive:
			ToggleLayoutModeAdaptive();
			break;
	}
}
```
We're in the process of supporting multiple icon sizes in all the layouts, this is just a rough concept but do you think something like this will work?

Reply:
If we always use `GridViewSize` regardless of the layout type, it seems fine.
We may need to add `force || (other conditions)` to all the `ifs` in the `switch`, but I need some testing to be sure about that. (Layouts may not be updated when changing the default one from settings)

Reply:
What does `force` do?

Reply:
> I would get rid of the first if statement

Still I think we should use the condition I wrote above, otherwise the layout would update only when increasing/decreasing icons' size, and not when the default layout is changed

Reply:
> What does `force` do?

When a user changes the default layout from settings, `LayoutMode` is updated in all the existing pages, but the layout itself is not applied. Because of that, `LayoutMode` would be equal to `DefaultLayoutMode` and so the layout wouldn't be updated. `force` is used to skip all the checks and force a layout refresh

Reply:
`LayoutMode is not FolderLayoutModes.LAYOUT_TYPE || GridViewSize != UserSettingsService.LayoutSettingsService.ICON_SIZE`
This should work but it'll need to be added to each case

Reply:
Should I implement it now?

Reply:
👍

Reply:
> > What does `force` do?
> 
> When a user changes the default layout from settings, `LayoutMode` is updated in all the existing pages, but the layout itself is not applied. Because of that, `LayoutMode` would be equal to `DefaultLayoutMode` and so the layout wouldn't be updated. `force` is used to skip all the checks and force a layout refresh

Nevermind, it's not needed using the new code